base/char.jl

```diff
@@ -50,18 +50,25 @@ Char
 (::Type{T})(x::AbstractChar) where {T<:Union{Number,AbstractChar}} = T(codepoint(x))
 (::Type{T})(x::T) where {T<:AbstractChar} = x
 
-codepoint(c::Char) = UInt32(c)
+"""
+    ncodeunits(c::Char) -> Int
 
+Return the number of code units required to encode a character as UTF-8.
+This is the number of bytes which will be printed if the character is written
+to an output stream, or `ncodeunits(string(c))` but computed efficiently.
 """
-    codepoint(c::AbstractChar)
+ncodeunits(c::Char) = max(1, 4 - (trailing_zeros(reinterpret(UInt32, c)) >> 3))
+
+"""
+    codepoint(c::AbstractChar) -> UInt32
```

Comment:
Perhaps `UInt32`->`Integer` if this is only for `Char` or add an extra line

```
codepoint(c::AbstractChar) -> Integer
codepoint(c::Char) -> UInt32
```

Reply:
Likewise, there's not reason not to represent a code point as a `UInt32`.

Reply:
Ok, I was thinking about the docs for this function that explicitly say

```
For `Char`, this is a `UInt32` value, but
`AbstractChar` types that represent only a subset of Unicode may
return a different-sized integer (e.g. `UInt8`).
```

The signature and the docs seems at odds now.

Reply:
Who wrote the docs are not really that important though... Just that the end result is consistent and doesn't say in the signature that `codepoint(c::AbstractChar)` has to return a `UInt32` and in the documentation just below it, that it can return a `UInt8`.